05/app.ts

```diff
@@ -0,0 +1,106 @@
+type CharacterType = 'Warrior' | 'Mage' | 'Healer';
+type CharacterSkill = 'Slam' | 'Fireball' | 'Heal';
+
+abstract class Character {
+    private name: string;
+    private _health: number;
+    public type: CharacterType;
+    public skill: CharacterSkill;
+
+    constructor(name: string, type: CharacterType, skill: CharacterSkill, health: number = 100) {
+        this.name = name;
+        this.type = type;
+        this._health = health;
+        this.skill = skill;
+    }
+
+    get health(): number {
+        return this._health;
+    }
+
+    set health(value: number) {
+        this._health = Math.max(0, Math.min(value, 100));
+        if (this._health === 0) {
+            console.log(`${this.name} umiera`);
+        }
+    }
+
+    getName(): string {
+        return this.name;
+    }
+
+    abstract useSkill(actor: Character, target: Character): void;
+
+    static performAction(actor: Character, target: Character) {
+        actor.useSkill(actor, target);
+    }
+}
+
+class WarriorClass extends Character {
```

Comment:
Nie ma sensu pisać Class bo to jest wiadome bez pisania jawnie. Raczej bym zrobił WarriorCharacter - często dodaje sie przyrostek lub przedrostek po nazwie klasy po której dziedziczymy.